Provide initial template for Gitlab Pipelines

snufkin opened the pull request:
- Adding an initial example gitlab CI configuration file.
- Adding Robo task that copies the template to the project root.

grasmash commented:
Nice!

grasmash commented on scripts/gitlab/gitlab-ci.yml:
```diff
@@ -0,0 +1,31 @@
+  stage: build
+  script:
+    - composer install
+    - blt setup:build
```
Any reason not to simply run `blt setup` ?

snufkin replied:
Well in my build step I'm actually composing a docker image, which contains just the static site build, without actually installing it. `blt setup` would also require mysql, and strictly speaking it would not belong to the build stage. (having a site install in there is a valid point though)

grasmash commented on scripts/gitlab/gitlab-ci.yml:
```diff
@@ -0,0 +1,31 @@
+  stage: test
+  script:
+    - blt validate
+
```
Seems like we should call `blt tests` here with some arguments like `--define tests.run-server=true`

snufkin replied:
In pipelines you define various build stages and then tasks for each stage. This particular example was to demonstrate that within the `test` stage you can have different task executions, e.g. in this one only linting happens.

grasmash commented:
I think this is incomplete. I'm still willing to merge it, but I'd like to add a warning after the command is executed:

`$this->logger->warning("Gitlab support is experimental and may not support all BLT features.");`

Or something like that.

snufkin commented:
Let me push some updates in the coming few workdays before you merge.

snufkin commented:
- Updated the template file to reflect correct invocations of BLT commands, based on the other CI examples.
- Added a validation step separately from testing so that CI would fail on CS violations.
- Added an extra warning about the template being experimental.
